### PR TITLE
Builtin support for GitHub Actions

### PR DESCRIPTION
**Pull request by fredrikekre**
Will clean up, and maybe split into more PRs, but it works: https://github.com/fredrikekre/Literate.jl/runs/247983206

### CODE REVIEW

**Comment by fredrikekre** [comment]
Okay, there was some bugs, but now it worked and deployed: https://github.com/fredrikekre/Literate.jl/runs/248626524

File: src/Documenter.jl [hunk 1]
```diff
@@ -500,6 +457,20 @@ function deploydocs(;
...
+            # outside root).
+            sha = try
+                readchomp(`git rev-parse --short HEAD`)
+            catch
+                # git rev-parse will throw an error and return code 128 if it is not being
+                # run in a git repository, which will make run/readchomp throw an exception.
+                # We'll assume that if readchomp fails it is due to this and set the sha
+                # variable accordingly.
+                "(not-git-repo)"
+            end
```
Comment (mortenpi) [inline src/Documenter.jl]: This conflicts with master?

Comment (fredrikekre) [reply src/Documenter.jl]: Yea, this PR is mostly a playground, review #1147 instead :p

**Comment by fredrikekre** [comment]
Pushed the interface for GH Actions here now.

**Pull request merged**